src/widget_selection.lisp

```diff
@@ -0,0 +1,159 @@
+		 :type unicode
+		 :initform (unicode "")
+		 :metadata (:sync t
+				  :json-name "button_style"
+				  :help "Use a predefined styling for the buttons. Options include: \"primary\", \"success\", \"info\", \"warning\", \"danger\", \"""\".")))
+  (:default-initargs
+   :view-name (unicode "ToggleButtonsView")
+    :model-name (unicode "ToggleButtonsModel"))
+  (:metaclass traitlets:traitlet-class))
+
+(defclass multiple-selection (%selection)
+  ((value :initarg :value :accessor value
+	  :type list
+	  :initform ()
+	  :metadata (:sync t
+			   :json-name "value"
+			   :help "Selected values"
+			   :to_json %values-to-labels
+			   :from_json %labels-to-values
+			   )))
+  (:metaclass traitlets:traitlet-class))
+
+(defclass dropdown (%selection)
+  ((button_style :initarg :button-style :accessor button-style
+		  :type unicode
+		  :initform (unicode "")
+		  :metadata (:sync t
+				   :json-name "button_style"
+				   :help "Use a predefined styling for the buttons.Options include:\"primary\", \"success\", \"info\", \"warning\", and \"danger\".")))
+  (:default-initargs
+   :view-name (unicode "DropdownView")
+    :model-name (unicode "DropdownModel"))
+  (:metaclass traitlets::traitlet-class))
+
+(defclass radio-buttons (%selection)
+  ()
+  (:default-initargs
+   :view-name (unicode "RadioButtonsView")
+    :model-name (unicode "RadioButtonsModel"))
+  (:metaclass traitlets:traitlet-class))
+
+(defclass select (%selection)
+  ()
+  (:default-initargs
+   :view-name (unicode "SelectView")
+    :model-name (unicode "SelectModel"))
+  (:metaclass traitlets:traitlet-class))
+
+(defclass selection-slider (%selection)
+  ((orientation :initarg :orientation :accessor orientation
+		 :type unicode
+		 :initform (unicode "horizontal")
+		 :metadata (:sync t
+				  :json-name "orientation"
+				  :help "vertical or horizontal"))
+   (readout :initarg :readout :accessor readout
+	     :type boolean
+	     :initform :true
+	     :metadata (:sync t
+			      :json-name "readout"
+			      :help "Display the current selected label next to the slider."))
+   (continuous_update :initarg :continuous_update :accessor continuous_update
+		       :type boolean
+		       :initform :true
+		       :metadata (:sync t
+					:json-name "continuous_update"
+					:help "Update the value of the widget as the user is holding the slider.")))
+  (:default-initargs
+   :view-name (unicode "SelectionSliderView")
+    :model-name (unicode "SelectionSliderModel"))
+  (:metaclass traitlets:traitlet-class))
+
+(defclass select-multiple (multiple-selection)
+  ()
+  (:default-initargs
+   :view-name (unicode "SelectMultipleView")
+    :model-name (unicode "SelectMultipleModel"))
+  (:metaclass traitlets:traitlet-class))
+
+(defun %label-to-value (k object)
+      (gethash 'k (slot-value object '_options_dict)))
+
+ (defun %labels-to-values (k obj)
+   (loop for v in k collect (%label-to-value v obj)))
+
+(defun %value-to-label (value obj)
+  (with-slots (options) obj
```

Comment:
This needs to do what the Python code does:
https://github.com/drmeister/spy-ipykernel/blob/master/ipywidgets/widgets/widget_selection.py#L16

(car (rassoc value (options obj) :test #'equal))